### PR TITLE
active tags on listings

### PR DESCRIPTION
**Pull request by michellejkwon**
simulator screen shot - iphone 7 - 2017-11-09 at 17 18 11
created active pickups view and navigation, along with tag component for listings (and associated styles) that appears only when listing is active

### CODE REVIEW

File: components/active_pickups/ActiveDetailScreen.js [hunk 1]
```diff
@@ -0,0 +1,162 @@
+import React from 'react';
```
Comment (amillman) [inline components/active_pickups/ActiveDetailScreen.js]: we actually probably dont need this screen. we can just use the `ListingDetailScreen` and register that as a screen for our `ActivePickupsNavigator`

Comment (amillman) [reply components/active_pickups/ActiveDetailScreen.js]: well also need the ability to cancel a pick up, which we can just make a red button instead of the green `Claim` button. and we will need to hit the API to transition from claimed to open

Comment (amillman) [reply components/active_pickups/ActiveDetailScreen.js]: we can check whether to display the red cancel or green claim button based on `this.props.listing.state`

File: components/active_pickups/ActivePickupsNavigator.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+const ActivePickupsNavigator = StackNavigator({
+  Active: {
+    screen: ActivePickupsScreen,
+    navigationOptions: ({navigation}) => ({
+      headerLeft: (Platform.OS === 'android') ? menuIcon(navigation) : null
+    }),
+  },
+  ActiveDetail: {
+    screen: NavigationHelper.paramsToProps(ActiveDetailScreen),
```
Comment (amillman) [inline components/active_pickups/ActivePickupsNavigator.js]: replace with `ListingDetailScreen`

**Comment by amillman** [comment]
@michellejkwon @nhiquach to solve that i added this package for listening to events. we already have a line in `ListingDetailScreen.js` under `_claim`, `window.EventBus.trigger(Events.listingClaimed, this.props.listing);` which will trigger any functions listening in on the event `listingClaimed` and pass in the listing object. to listen in on an event, we can call `window.EventBus.on(Events.listingClaimed, _someFunc)`, where `_someFunc` will take in a listing as a parameter. We need to put this in `componentDidMount` to start listening, and also call `window.EventBus.off(Events.listingClaimed, _someFunc)` in `componentWillDismount` to stop listening

File: constants/ComponentStyles.js [hunk 3]
```diff
@@ -34,7 +34,25 @@ const ComponentStyles = StyleSheet.create({
     zIndex: -10,
     flex: 1,
     resizeMode: 'cover',
-  }
+  },
+
+  tagContainer: {
+    position: 'absolute',
```
Comment (amillman) [inline constants/ComponentStyles.js]: remove this

File: constants/ComponentStyles.js [hunk 4]
```diff
@@ -34,7 +34,25 @@ const ComponentStyles = StyleSheet.create({
...
+    borderRadius: 2,
+    bottom: 10,
+    right: 10,
+    width: 60,
+    height: 20,
```
Comment (amillman) [inline constants/ComponentStyles.js]: dont harcode width height T_T use `paddingHorizontal` and `paddingVertical` instead

File: constants/ComponentStyles.js [hunk 5]
```diff
@@ -34,7 +34,25 @@ const ComponentStyles = StyleSheet.create({
...
+    backgroundColor: Colors.blue,
+  },
+
+  tagText: {
+    marginTop: 3,
```
Comment (amillman) [inline constants/ComponentStyles.js]: put this as `paddingVertical` value instead of hardcoding height

File: requesters/ListingsRequester.js [hunk 6]
```diff
@@ -12,13 +13,29 @@ class ListingsRequester extends BaseRequester {
     return BaseRequester.get(APIConstants.listings.all, params);
   }
 
+  static async getActiveListings() {
+    params = {};
+    params.state = "claimed";
+    try {
+      user = await LocalStorage.getUser();
+      params.marketplace_user_id = user.id;
```
Comment (nhiquach) [inline requesters/ListingsRequester.js]: I think we also have a scope for state for you should be able to add this to params as well. https://github.com/replate/replate-business/blob/master/app/controllers/api/marketplace/marketplace_listings_controller.rb

File: requesters/ListingsRequester.js [hunk 7]
```diff
@@ -12,13 +13,29 @@ class ListingsRequester extends BaseRequester {
     return BaseRequester.get(APIConstants.listings.all, params);
   }
 
+  static async getActiveListings() {
+    params = {};
+    params.state = "claimed";
```
Comment (nhiquach) [inline requesters/ListingsRequester.js]: can we make "open" and "claimed" constants? something like from the second answer https://stackoverflow.com/questions/39036457/react-create-constants-file and it can be declared in this file at the top

Comment (nhiquach) [reply requesters/ListingsRequester.js]: or even like one of our Constants files

Comment (amillman) [reply requesters/ListingsRequester.js]: @michellejkwon look at `Events.js` and you can make something similar like `ModelConstants.js`. We eventually want to be able to call something like `ModelConstants.Listing.state.CLAIMED`

File: components/listings/ListingsScreen.js [hunk 8]
```diff
@@ -32,7 +32,12 @@ class ListingsScreen extends React.Component {
 
   componentDidMount() {
     this._getListings();
-    window.EventBus.on(Events.listingClaimed, this._getListings());
+    window.EventBus.on(Events.listingClaimed, this._removeListing);
```
Comment (amillman) [inline components/listings/ListingsScreen.js]: we dont need an event bus here, because were already calling `_removeListing` as a callback from the `ListingDetailScreen`

File: constants/ComponentStyles.js [hunk 9]
```diff
@@ -38,16 +38,15 @@ const ComponentStyles = StyleSheet.create({
 
   tagContainer: {
     position: 'absolute',
-    borderRadius: 2,
-    bottom: 10,
-    right: 10,
-    width: 60,
-    height: 20,
+    borderRadius: UIConstants.cornerRadii.standard,
```
Comment (amillman) [inline constants/ComponentStyles.js]: dont use standard radius here, because this is very specific to the tag. 2 was fine

Comment (amillman) [inline constants/ComponentStyles.js]: also i think you should make a new component called `Tag.js` and style it in there, cuz well probably need this tag for other places. let the component take in a prop called `text` a string and `color` also a string

File: constants/ComponentStyles.js [hunk 10]
```diff
@@ -38,16 +38,15 @@ const ComponentStyles = StyleSheet.create({
...
+    bottom: UIConstants.margins.standard,
+    right: UIConstants.margins.standard,
+    padding: UIConstants.margins.tight,
+    paddingVertical: '0.3%',
```
Comment (amillman) [inline constants/ComponentStyles.js]: hardcode the paddingVertical, dont use percentage

File: components/listings/ListingDetailScreen.js [hunk 11]
```diff
@@ -66,6 +79,22 @@ class ListingDetailScreen extends React.Component {
...
 
   render() {
+    let button = (
+      <LoadingButton
+      containerStyle={[ComponentStyles.buttonContainer, styles.claimButtonContainer]}
+      style={[ComponentStyles.buttonText, styles.claimButtonText]}
+      onPress={this._claim}
+      isLoading={this.state.isClaiming}
+      title="Claim" />)
+    if (this.props.listing.state == "claimed") {
```
Comment (nhiquach) [inline components/listings/ListingDetailScreen.js]: change to ModelConstants.listing.state.CLAIMED

File: components/common/Tag.js [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+
+class Tag extends React.Component {
+
+  static propTypes = {
+    text: PropTypes.string,
+    color: PropTypes.string,
+  }
+
+  render() {
+    console.log(this.props);
```
Comment (nhiquach) [inline components/common/Tag.js]: nit: remove

File: components/listings/ListingsScreen.js [hunk 13]
```diff
@@ -13,6 +13,7 @@ import ListingsRequester from '../../requesters/ListingsRequester';
 
 import Events from '../../constants/Events';
 
+
```
Comment (nhiquach) [inline components/listings/ListingsScreen.js]: nit: remove unnecessary whitespace

File: components/active_pickups/ActivePickupsScreen.js [hunk 14]
```diff
@@ -0,0 +1,118 @@
...
+      listings: newListings,
+    });
+  }
+
+  _keyExtractor = (item, index) => {
+    return item.id;
+  }
+
+  _onPressItem = (listing) => {
+    this.props.navigation.navigate('ActiveDetail', {listing: listing, onClaim: this._removeListing});
```
Comment (amillman) [inline components/active_pickups/ActivePickupsScreen.js]: rename `onClaim` to `onCancel`

File: components/listings/ListingItem.js [hunk 15]
```diff
@@ -19,13 +20,18 @@ class ListingItem extends React.Component {
   static propTypes = {
     listing: PropTypes.object,
     onPressItem: PropTypes.func,
+    active: PropTypes.bool,
```
Comment (amillman) [inline components/listings/ListingItem.js]: right below this, add:

```static defaultProps = {
    active: false```

File: components/listings/ListingsScreen.js [hunk 16]
```diff
@@ -84,6 +98,7 @@ class ListingsScreen extends React.Component {
               <ListingItem
                 listing={item}
                 onPressItem={this._onPressItem}
+                active={false}
```
Comment (amillman) [inline components/listings/ListingsScreen.js]: remove since well default active to false

File: components/active_pickups/ActivePickupsScreen.js [hunk 17]
```diff
@@ -0,0 +1,118 @@
...
+    return (
+      <LoadingView style={styles.container} isLoading={this.state.isLoading}>
+        <FlatList
+            data={this.state.listings}
+            keyExtractor={this._keyExtractor}
+            renderItem={ ({item}) =>
+              <ListingItem
+                listing={item}
+                onPressItem={this._onPressItem}
+                active={true}
```
Comment (amillman) [inline components/active_pickups/ActivePickupsScreen.js]: for booleans, we can actually just say `active` and react will know that we want to set it to true
